ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -471,22 +471,27 @@ class Endpoints(
 
   private def httpResponse[A: JsonWriter](
       result: ET[domain.SyncResponse[A]]
-  ): Future[HttpResponse] = {
-    val fa: Future[Error \/ (JsValue, StatusCode)] =
-      result.flatMap { x =>
-        either(SprayJson.encode1(x).map(y => (y, x.status)).liftErr(ServerError))
-      }.run
-    fa.map {
-      case -\/(e) =>
-        httpResponseError(e)
-      case \/-((jsVal, status)) =>
-        HttpResponse(
-          entity = HttpEntity.Strict(ContentTypes.`application/json`, format(jsVal)),
-          status = status,
-        )
-    }.recover { case NonFatal(e) =>
-      httpResponseError(ServerError(e.description))
-    }
+  )(implicit metrics: Metrics): Future[HttpResponse] = {
+    Timed.future(
+      metrics.daml.HttpJsonApi.responseCreationTimer,
+      result
+        .flatMap { x =>
+          either(SprayJson.encode1(x).map(y => (y, x.status)).liftErr(ServerError))
+        }
+        .run
+        .map {
+          case -\/(e) =>
+            httpResponseError(e)
+          case \/-((jsVal, status)) =>
+            HttpResponse(
+              entity = HttpEntity.Strict(ContentTypes.`application/json`, format(jsVal)),
+              status = status,
+            )
+        }
+        .recover { case NonFatal(e) =>
+          httpResponseError(ServerError(e.description))
```

Comment:
I understand that this comes from existing code, but should we also take advantage of this chance to log this exception?

Reply:
I am generally against logging anything that is reported by proper channels already; it's a good way to leak sensitive data.

Reply:
We are recovering from a server error here, I think it makes sense for the information to be logged on the server. I would argue it would probably make less sense to return the full description to the client (e.g. if the connection to the database fails I'd be likely be more interested to have it logged rather than reported to the client). Does that make sense?

Reply:
If you can honestly say "don't report the description to the client at all in this case", then it makes sense. Otherwise, we're pointing in one direction and jumping in the other.

Reply:
If the errors caught here are meaningful to be reported in full only to the operator of the server, then yes. There are cases in which both a detailed report to the user _and_ logging to the advantage of the operator could make sense. Not sure of what could be thrown here. It looks to me like here most exceptions would be interesting from an operational point of view but less so from a user perspective, but I'll let you judge that.

Reply:
This escalated quickly :joy:

Thing is, how do we know whether the information is either best only for the user or the operator of the server?

Reply:
I want to postpone this to a seperate pr and better discuss this in slack or in an gh issue